cli/src/tests/parser_test.rs

```diff
@@ -872,6 +872,48 @@ fn test_parsing_with_multiple_included_ranges() {
     );
 }
 
+#[test]
+fn test_parsing_with_included_range_at_the_beginning_of_file() {
+    let source_code = "<div>test</div>{_ignore_this_part_}";
+
+    let mut parser = Parser::new();
+    parser.set_language(get_language("html")).unwrap();
+
+    let end_byte = source_code.find("{_ignore_this_part_").unwrap();
+
+    let range_to_parse = Range {
+        start_byte: 0,
+        start_point: Point {
+            row: 10,
+            column: 12,
+        },
+        end_byte,
+        end_point: Point {
+            row: 10,
+            column: 12 + end_byte,
+        },
+    };
+
+    parser.set_included_ranges(&[range_to_parse]).unwrap();
+
+    let mut callback = |offset, _point| {
+        if offset == 0 {
+            source_code.as_bytes()
+        } else {
+            &[]
+        }
+    };
```

Comment:
Is the callback important here? Could you just pass the `source_code.as_bytes()` directly to `parser.parse`?

Reply:
It is still possible to reproduce the issue by passing the bytes directly.  However because the bytes and position are mismatched, it seemed like it was more appropriate to use the callback instead.  Since it's basically saying here's a piece of the file instead of the entire file (but maybe that's not an important distinction).

Reply:
I'd rather have this test just use the bytes. I think that the use of a callback vs a byte slice can be considered orthogonal to whether the points match the byte offsets. The callback is more needed when we have the text stored in some non-contiguous structure.